### PR TITLE
Fix `lib_tool.read_to_keys`

### LINKED ISSUE #1835: `lib_tool.read_to_keys` doesn't work for some `KeyType`s
### Describe the bug

`lib_tool.read_to_keys` crashes for `KeyType.VERSION_REF`

`lib_tool.read_to_keys` gives wrong index keys when reading `KeyType.VERSION`

There are maybe other problematic key types but we lack testing. We need to add thorough tests that verify we can hop around the version chain without issues over all key types.

### Steps/Code to Reproduce

```
from arcticdb import Arctic
from arcticdb.toolbox.library_tool import KeyType
import pandas as pd

ac = Arctic("lmdb://data/arcticdb")
ac.delete_library("vel_test")
lib = ac.get_library("vel_test", create_if_missing=True)
lib_tool = lib._nvs.library_tool()
sym = "sym"

# Write a single version
df = pd.DataFrame({"data": [1]}, index=[pd.Timestamp("2024-09-09")])
lib.write(sym, df)

# Fails to read ref key
ref_key = lib_tool.find_keys_for_id(KeyType.VERSION_REF, sym)[0]
try:
    lib_tool.read_to_keys(ref_key)
except Exception as e:
    print(e) # This gives "std::bad_variant_access(std::get: wrong index for variant)"

# Reads wrong index keys from version key
ver_key = lib_tool.find_keys_for_id(KeyType.VERSION, sym)[0]
print(lib_tool.read_to_keys(ver_key)) # This gives "[i:sym:0:0xe79e1d6ff40aaa66@1726559612750387819[1725840000,1725840000000000001]]" which has an obviously wrong start_index
```

### Expected Results

`lib_tool.read_to_keys` correctly read all key types which contain keys (`VERSION_REF, VERSION, INDEX`)
`lib_tool.read_to_dataframe` works on all key types.

### OS, Python Version and ArcticDB Version

Every version.

### Backend storage used

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by IvoDD**
Also adds tests showcasing that iterating over version ref, version, index and data key now works

#### Reference Issues/PRs
Fixes #1835 

#### Checklist

<details>
  <summary>
   Checklist for code changes...
  </summary>
 
 - [ ] Have you updated the relevant docstrings, documentation and copyright notice?
 - [ ] Is this contribution tested against [all ArcticDB's features](../docs/mkdocs/docs/technical/contributing.md)?
 - [ ] Do all exceptions introduced raise appropriate [error messages](https://docs.arcticdb.io/error_messages/)?
 - [ ] Are API changes highlighted in the PR description?
 - [ ] Is the PR labelled as enhancement or bug so it appears in autogenerated release notes?
</details>

### CODE REVIEW

**Review by G-D-Petrov — COMMENTED** [review]
There is already a file for library_tool tests:
`ArcticDB/python/tests/integration/toolbox/test_library_tool.py`

**Comment by IvoDD** [comment]
This fixes one part of #1835 
We still need more thorough testing for all key types.

**Comment by IvoDD** [comment]
Now fixes #1835 entirely.

File: python/tests/integration/toolbox/test_library_tool.py [hunk 1]
```diff
@@ -137,3 +138,83 @@ def test_count_keys(object_and_mem_and_lmdb_version_store):
...
+
+    df = pd.DataFrame({"a": [1, 2, 3]})
+    if use_time_index:
+        df = pd.DataFrame(index=pd.date_range(start=pd.Timestamp(0), periods=3), data={"a": [1, 2, 3]})
+    lib.write(sym, df)
+
+    ver_ref_key = lib_tool.find_keys_for_symbol(KeyType.VERSION_REF, sym)[0]
+    ver_key = lib_tool.read_to_keys(ver_ref_key)[-1]
+    index_key = lib_tool.read_to_keys(ver_key)[0]
+    data_key = lib_tool.read_to_keys(index_key)[0]
```
Comment (poodlewars) [inline python/tests/integration/toolbox/test_library_tool.py:157]: Assertions on the structure of these 3 keys, and checks that they can be used to read segments

File: python/tests/integration/toolbox/test_library_tool.py [hunk 2]
```diff
@@ -137,3 +138,83 @@ def test_count_keys(object_and_mem_and_lmdb_version_store):
     assert lib_tool.count_keys(KeyType.SNAPSHOT_REF) == 1
     assert lib_tool.count_keys(KeyType.MULTI_KEY) == 1
     assert lib_tool.count_keys(KeyType.SNAPSHOT) == 0
+
+
+@pytest.mark.parametrize("use_time_index", [True, False])
+def test_read_data_key_from_version_ref(object_and_mem_and_lmdb_version_store, use_time_index):
```
Comment (poodlewars) [inline python/tests/integration/toolbox/test_library_tool.py:144]: Why `object_and_mem_and_lmdb_version_store`? Just `mem` is probably fine

File: python/tests/integration/toolbox/test_library_tool.py [hunk 3]
```diff
@@ -137,3 +138,83 @@ def test_count_keys(object_and_mem_and_lmdb_version_store):
...
+def test_iterate_version_chain_with_lib_tool(object_and_mem_and_lmdb_version_store):
+    lib = object_and_mem_and_lmdb_version_store
+    lib_tool = lib.library_tool()
+    sym = "sym"
+    num_versions = 20
+
+    # Populate some versions
+    df = pd.DataFrame(index=pd.date_range(start=pd.Timestamp(0), periods=3), data={"a": [1, 2, 3]})
+    for i in range(num_versions):
+        prune_previous = i%3 == 0
```
Comment (poodlewars) [inline python/tests/integration/toolbox/test_library_tool.py:177]: (nit) `i % 3` not `i%3`

File: python/tests/integration/toolbox/test_library_tool.py [hunk 4]
```diff
@@ -137,3 +138,83 @@ def test_count_keys(object_and_mem_and_lmdb_version_store):
...
+            try:
+                next_keys = lib_tool.read_to_keys(key)
+            except:
+                # Deleted index key
+                next_keys = []
+        for next_key in next_keys:
+            iterate_through_version_chain(next_key)
+
+    version_ref = lib_tool.find_keys_for_symbol(KeyType.VERSION_REF, sym)[0]
+    print(version_ref)
```
Comment (poodlewars) [inline python/tests/integration/toolbox/test_library_tool.py:206]: remove

File: python/tests/integration/toolbox/test_library_tool.py (hunk 2 above)
Comment (IvoDD) [reply python/tests/integration/toolbox/test_library_tool.py]: Ah was just copying what the other fixtures did. Agreed that these are specifically testing the library tool functionality so mem is just fine.

**Review by poodlewars — APPROVED** [review]

**Pull request merged**